### ISSUE TITLE
Glossary for technical terms

### ISSUE THREAD

**Issue by TomKellyGenetics** [swcarpentry-ja/i18n#23]
Technical terms are used across lessons and need to be translated consistently between them to avoid confusion. A "glossary" will be set up to document frequently used technical terms in the repo Wiki.

If there is a choice between a Katakana (English loanword) form or a Japanese word in Kanji, the Katakana form will be preferred. This has the advantage of being similar to the English terms on forums such as Stack Exchange and avoiding archaic terms that aren't used very often. Hopefully this will guide consistent translations between lessons. We will use lists used for book translation as a starting point.

There are several exceptions:

-  Where another term is commonly used by Japanese in this field. This will need the input of Japanese people who have experience with these concepts and tools.

- Where an English term should be explained rather than translated because it is easier to understand in context #21

**Comment by TomKellyGenetics** [comment]
The vocabulary in the glossary for the Spanish version of Software Carpentry should also be included:

https://github.com/Carpentries-ES/board/blob/master/Convenciones_Traduccion.md

**Comment by TomKellyGenetics** [comment]
It also needs to be considered how to handle abbreviations and terms that are commonly abbreviated (such as CLI or IDE). Should these be translated or kept as abbreviation?

**Comment by joelnitta** [comment]
I don't think we should one approach across the board, and would rather leave it up to the translators. Some terms are commonly used as abbreviations in Japanese, whereas others have an equivalent in kanji or kana. Either way, I think it will be helpful to explain what each word in the English abbreviation means the first time it's used (as in [issue 21](https://github.com/swcarpentry-ja/i18n/issues/21)). I think it is more important that we are consistent once a choice is made for a particular term.

Actually, the Spanish guide is useful here*. They have a "list of terms to translate" and a 
"list of terms NOT to translate". So a given acronym might be in either list. The important thing is that everyone treats each term the same.

Since we are still getting started on the glossary, my suggestion for now is to include the English term and its definition, but without a fixed Japanese translation yet. Then, the first translator to start using a specific term should enter their translation on the glossary, and others should use it. If there are disagreements, they should be flagged as issues and resolved.

*On a related note, it looks like many of the Spanish "list of terms NOT to translate" are on that list because they are also commands or used in code. I agree that we should avoid translating these, and instead explain the meaning the first time one is used.

**Comment by TomKellyGenetics** [comment]
Yeah that’s basically what I had in mind. I was actually wondering if we want a separate column for abbreviations so we can include the Japanese word or definition as well.

**Comment by joelnitta** [comment]
Including columns for both English and Japanese short definitions would probably be beneficial for everyone, as long as they aren't too burdensome to write.

@TomKellyGenetics I recall you had started on a glossary. Do you want to submit a PR for that so we can start working off it? Like I said, I think it's fine if it doesn't have any Japanese translations yet. Just agreeing on what words should be included in the glossary and translated (or not) is an important step.

I think this is an issue we should try to get resolved ASAP since we already have translators starting to contribute.

**Comment by TomKellyGenetics** [comment]
It's already in the Wiki tab (work in progress)

https://github.com/swcarpentry-ja/i18n/wiki/Glossary-for-technical-terms

**Comment by joelnitta** [comment]
Whoops, missed that. I always felt like github wikis are a bit buried... I'll go ahead and add the link to the guidelines are merge it.

**Comment by TomKellyGenetics** [comment]
This seems to be rarely used but it may still be useful for maintaining the lessons later on.

Another option is to use the glosario project instead:
https://github.com/carpentries/glosario
